src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseHybridFVMAssemblerHelperKernels.hpp

```diff
@@ -0,0 +1,673 @@
+                          ElementViewConst< arrayView4d< real64 const > > const & dPhaseMassDens_dCompFrac,
+                          ElementViewConst< arrayView2d< real64 const > > const & phaseMob,
+                          ElementViewConst< arrayView2d< real64 const > > const & dPhaseMob_dPres,
+                          ElementViewConst< arrayView3d< real64 const > > const & dPhaseMob_dCompDens,
+                          ElementViewConst< arrayView3d< real64 const > > const & dCompFrac_dCompDens,
+                          ElementViewConst< arrayView4d< real64 const > > const & phaseCompFrac,
+                          ElementViewConst< arrayView4d< real64 const > > const & dPhaseCompFrac_dPres,
+                          ElementViewConst< arrayView5d< real64 const > > const & dPhaseCompFrac_dCompFrac,
+                          ElementViewConst< arrayView1d< globalIndex const > > const & elemDofNumber,
+                          arraySlice2d< real64 const > const & transMatrixGrav,
+                          real64 const (&oneSidedVolFlux)[ NF ],
+                          real64 const (&dOneSidedVolFlux_dPres)[ NF ],
+                          real64 const (&dOneSidedVolFlux_dFacePres)[ NF ][ NF ],
+                          real64 const (&dOneSidedVolFlux_dCompDens)[ NF ][ NC ],
+                          real64 const & dt,
+                          CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                          arrayView1d< real64 > const & localRhs )
+  {
+    localIndex constexpr NDOF = NC+1;
+
+    // dof numbers
+    globalIndex dofColIndicesElemVars[ NDOF*(NF+1) ] = { 0 };
+    globalIndex dofColIndicesFaceVars[ NF ] = { 0 };
+    for( localIndex idof = 0; idof < NDOF; ++idof )
+    {
+      dofColIndicesElemVars[idof] = elemDofNumber[localIds[0]][localIds[1]][localIds[2]] + idof;
+    }
+
+    // divergence of fluxes
+    real64 divMassFluxes[ NC ] = { 0.0 };
+    real64 dDivMassFluxes_dElemVars[ NC ][ NDOF*(NF+1) ] = {{ 0.0 }};
+    real64 dDivMassFluxes_dFaceVars[ NC ][ NF ] = {{ 0.0 }};
+
+    // auxiliary variables for upwinding
+
+    // upwinding phase buoyancy transport coefficients
+    real64 upwPhaseViscCoef[ NP ][ NC ] = {{ 0.0 }};
+    real64 dUpwPhaseViscCoef_dPres[ NP ][ NC ] = {{ 0.0 }};
+    real64 dUpwPhaseViscCoef_dCompDens[ NP ][ NC ][ NC ] = {{{ 0.0 }}};
+    globalIndex upwViscDofNumber = 0;
+
+    // gravity term: ( \rho_l - \rho_m ) g \Delta z
+    real64 phaseGravTerm[ NP ][ NP-1 ] = {{ 0.0 }};
+    real64 dPhaseGravTerm_dPres[ NP ][ NP-1 ][ 2 ] = {{{ 0.0 }}};
+    real64 dPhaseGravTerm_dCompDens[ NP ][ NP-1 ][ 2 ][ NC ] = {{{{ 0.0 }}}};
+
+    // upwinding phase buoyancy transport coefficients
+    real64 upwPhaseGravCoef[ NP ][ NP-1 ][ NC ] = {{{ 0.0 }}};
+    real64 dUpwPhaseGravCoef_dPres[ NP ][ NP-1 ][ NC ][ 2 ] = {{{{ 0.0 }}}};
+    real64 dUpwPhaseGravCoef_dCompDens[ NP ][ NP-1 ][ NC ][ 2 ][ NC ] = {{{{{ 0.0 }}}}};
```

Comment:
You can also zero-initialize an array via just a single `{}` regardless of number of dimensions: https://godbolt.org/z/9Ex17d
Of course it's up to you if you'd like initializers to be more explicit.

Reply:
I was not aware of this, but the syntax is much lighter index, I made the modification.